GRA.5 and 6 rework

jni2000 opened the pull request:
Dear reviewers

We understand that GRA.5 and 6 depends on FAD.2 approval, which is currently in progress. Please help review other parts in parallel with the review FAD.2. We will do the final adjustment to and related changes required for FAD.2 when it is approved.

Thanks

jni2000 commented:
Kate

We have update GRA.5 and 6 based on the newest development on FAD.2. I think it is time to review these two now. Please review. Thanks.

kate-harrison commented:
I'm sorry, but I've said many times that I think the most efficient course of action is to review and submit the FAD.2 and SAS TH code and merge that into this PR before reviewing this PR. I still intend to follow this course of action.

jni2000 commented:
Yes, let's focus on FAD.2  and get it in as soon as possible. We are updating these according to the new revision in FAD.2. We want to keep up the speed ourselves. We will definitely do the final catch up on GRA5 and 6 once FAD.2 is finalized. Don't worry, please.

________________________________
From: Kate <notifications@github.com>
Sent: Friday, March 2, 2018 12:00:12 PM
To: Wireless-Innovation-Forum/Spectrum-Access-System
Cc: James Ni; Author
Subject: Re: [Wireless-Innovation-Forum/Spectrum-Access-System] GRA.5 and 6 rework (#440)


I'm sorry, but I've said many times that I think the most efficient course of action is to review and submit the FAD.2 and SAS TH code and merge that into this PR before reviewing this PR. I still intend to follow this course of action.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Wireless-Innovation-Forum/Spectrum-Access-System/pull/440#issuecomment-369982960>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AKpZrFt7rOJeuRr45R2s9Mqt1DwSyBLpks5taXqcgaJpZM4SUUC4>.

jni2000 commented:
Kate & Other reviewers

We have updated GRA.5 and 6 based on the approved FAD.2 and ready for review. Please take a look. Thanks.

kate-harrison commented on src/harness/testcases/WINNF_FT_S_GRA_testcase.py:
```diff
@@ -276,6 +276,232 @@ def test_WINNF_FT_S_GRA_4(self):
+    del device_c2['cbsdCategory']
+    del device_c2['airInterface']
+    del device_c2['installationParam']
+
```
Do you also need to delete `measCapability`?

jni2000 replied:
deleted

kate-harrison commented on src/harness/testcases/WINNF_FT_S_GRA_testcase.py:
```diff
@@ -276,6 +276,232 @@ def test_WINNF_FT_S_GRA_4(self):
+        'sasTestHarnessName': 'SAS-Test-Harness-1',
+        'hostName': 'localhost',
+        'port': 9001,
+        'serverCert': "certs/server.cert",
```
Use os.path.join() instead so that this will work on other OSes.

jni2000 replied:
changed as suggested

kate-harrison commented on src/harness/testcases/WINNF_FT_S_GRA_testcase.py:
```diff
@@ -276,6 +276,232 @@ def test_WINNF_FT_S_GRA_4(self):
+    grant_g2['operationParam']['operationFrequencyRange'][
+        'highFrequency'] = 3655000000
+
+    cbsd_record = [device_c1]
```
Please use plural to indicate that these are lists.

jni2000 replied:
changed to plural for all applicable cases.

jni2000 commented:
Kate, thanks for the review comments. We have made necessary revision as you suggested. Please review.

Regarding fake_sas.py changes: 
GetSasImplementationRecord method is not overrided in fakeSAS handler in master version but it was defined in sas_interface.py 
that cause fake_sas stopped working and not able to register devices. So we have fixed this defect in fake_sas.py in order to test GRA_5 and  GRA_6

kate-harrison commented on src/harness/testcases/WINNF_FT_S_GRA_testcase.py:
```diff
@@ -314,21 +315,20 @@ def generate_GRA_5_default_config(self, filename):
     }
     sas_harness_dump_records = {
-        'cbsdRecords': generateCbsdRecords(cbsd_record, [grant_record])
+        'cbsdRecords': generateCbsdRecords(cbsd_records, [grant_records])
```
Hm, I'm now seeing that my earlier suggestion was not the best: it's now not very clear why `grant_records` is being wrapped in a list while `cbsd_records` is not. One way to fix it would be to do the following:

```
  'cbsdRecords': generateCbsdRecords([device_c1], [ [grant_g1] ])
```

since you don't actually use `cbsd_records` and `grant_records` anywhere else. Then it's a bit clearer what's going on. Sorry for the back-and-forth!

jni2000 replied:
Updated   accordingly

kate-harrison commented on src/harness/testcases/WINNF_FT_S_GRA_testcase.py:
```diff
@@ -486,18 +488,17 @@ def test_WINNF_FT_S_GRA_6(self, config_filename):
 
-    # As Python garbage collector is not very consistent, directory is not getting deleted. 
+    # As Python garbage collector is not very consistent, directory is not getting deleted.
     # Hence, explicitly stopping SAS Test Hanress and cleaning up
```
Typo: "Harness"

jni2000 replied:
Typo has   been fixed now

kate-harrison commented on src/harness/testcases/configs/test_WINNF_FT_S_GRA_5/default.config:
```diff
@@ -29,11 +29,11 @@
           }
         ],
-        "id": "cbsd/test_fcc_id_a/d7a9fe1be84243ebdd50c1359cf0630c3d273350",
+        "id": "cbsd/cbsd/test_fcc_id_a/d7a9fe1be84243ebdd50c1359cf0630c3d273350",
```
Hmm, I just noticed that this is getting prepended twice. Probably a small bug in `generateCbsdRecords()`.

jni2000 replied:
Fixed   the bug in generateCbsdRecords()

kate-harrison commented on src/harness/fake_sas.py:
```diff
@@ -182,6 +182,16 @@ def Deregistration(self, request, ssl_cert=None, ssl_key=None):
         })
     return response
 
+  def GetSasImplementationRecord(self, request, ssl_cert=None, ssl_key=None):
```
> Regarding fake_sas.py changes:
> GetSasImplementationRecord method is not overrided in fakeSAS handler in master version but it was defined in sas_interface.py
> that cause fake_sas stopped working and not able to register devices. So we have fixed this defect in fake_sas.py in order to test GRA_5 and GRA_6


You're right, that method was taken out of the fake SAS in https://github.com/Wireless-Innovation-Forum/Spectrum-Access-System/pull/458 . It's my fault that we didn't also remove it from sas_interface.py and sas.py, which is why you're running into problems. We'll issue a PR to fix this ASAP, and then I think you should be able to revert the changes in this file.

jni2000 replied:
Ok, we have removed this fix in PR440

jni2000 commented:
Kate, we have made necessary fixes/changes for GRA.5 and 6 per your review comments. Please take a look.

Thanks.

kate-harrison commented on src/harness/fake_sas.py:
```diff
@@ -276,6 +276,9 @@ def InjectWisp(self, request):
   def InjectSasAdministratorRecord(self, request):
     pass
 
+  def InjectSasImplementationRecord(self, request):
```
Please remove.

jni2000 replied:
removed

jni2000 commented:
InjectSasImplementationRecord removed, please continue review. Thanks.

kate-harrison commented on src/harness/sas_test_harness.py:
```diff
@@ -345,7 +345,7 @@ def generateCbsdRecords(registration_requests, grant_requests_list):
 
     internal_data = {
-        'id': 'cbsd/{}'.format(cbsd_reference_id),
+        'id': '{}'.format(cbsd_reference_id),
```
Can't you just use `cbsd_reference_id` directly?

jni2000 replied:
yes, you are right. changed as suggested.

jni2000 commented:
Thanks, Kate. Removed the default config files and changed below line to use cbsd_reference_id directly as you suggested. Please do a final check. Thanks.

'id': '{}'.format(cbsd_reference_id),
